gather_istio.sh

```diff
@@ -0,0 +1,244 @@
+
+  # set global variable which is used when calling 'oc adm inspect'
+  get_log_collection_args
+
+  # TODO: add new name label to servicemesh-operator3 pod and use that instead of the control-plane label
+  operatorNamespace=$(oc get pods --all-namespaces -l control-plane=servicemesh-operator3 -o jsonpath="{.items[0].metadata.namespace}")
+  # this gets also logs for all pods in that namespace
+  inspect "ns/$operatorNamespace"
+  inspect clusterserviceversion "${operatorNamespace}"
+
+  inspect nodes
+
+  # TODO: we should probably add a new label or start using 'install.operator.istio.io/owning-resource'
+  # another option is to use 'metadata.ownerReferences' but having a label would be easier for a query
+  for r in $(oc get clusterroles,clusterrolebindings -l install.operator.istio.io/owning-resource -oname); do
+    inspect "$r"
+  done
+  for r in $(oc get clusterroles,clusterrolebindings -l 'app in (istiod,istio-reader)' -oname); do
+    inspect "$r"
+  done
+
+  # inspect all istio.io CRDs
+  crds="$(getCRDs)"
+  for crd in ${crds}; do
+    inspect "crd/${crd}"
+  done
+
+  # inspect all controlled mutatingwebhookconfiguration
+  for mwc in $(oc get mutatingwebhookconfiguration -l app=sidecar-injector -o name);do
+    inspect "${mwc}"
+  done
+  # inspect all controlled validatingwebhookconfiguration
+  for vwc in $(oc get validatingwebhookconfiguration -l app=istiod -o name);do
+    inspect "${vwc}"
+  done
+
+  # inspect Istio and IstioRevision resources, this will just store all CRs as those are cluster-scoped resources
+  inspect "Istio"
+  inspect "IstioRevision"
+
+  # inspect all namespaces with Istio components
+  controlPlanes=$(oc get IstioRevision -o custom-columns=NAME:spec.namespace --no-headers | sort -u)
+  for cp in ${controlPlanes}; do
+    echo
+    echo "Processing control plane namespace: ${cp}"
+
+    inspectNamespace "$cp"
+  done
+
+  # iterate over all Istio revisions
+  for ir in $(oc get IstioRevision -o jsonpath="{.items[*].metadata.name}"); do
+    echo
+    echo "Inspecting ${ir} IstioRevision"
+    cpNamespace=$(oc get IstioRevision "${ir}" -o jsonpath="{.spec.namespace}")
+
+    getSynchronization "${cpNamespace}" "${ir}"
+
+    # iterate over all namespaces which have pods with proxy pointing to this control plane revision
+    # TODO: this might be very resource demanding and causing performance problems in clusters with a lot of pods
+    for dpn in $(oc get pods -A -o=jsonpath="{.items[?(@.metadata.annotations.istio\.io/rev==\"${ir}\")].metadata.namespace}" | tr ' ' '\n' | sort -u); do
+      echo
+      echo "Inspecting ${dpn} data plane namespace"
+      inspectNamespace "${dpn}"
+      getEnvoyConfigForPodsInNamespace "${cpNamespace}" "${ir}" "${dpn}"
+    done
+
+    istioCniNamespace=$(oc get IstioCNI -A -o jsonpath="{.items[0].spec.namespace}")
+    if [ -n "$istioCniNamespace" ]
+    then
+      inspectNamespace "${istioCniNamespace}"
+      # cluster-scoped resource
+      inspect "IstioCNI"
```

Comment:
Ah, you inspect IstioCNI here. But wouldn't it be cleaner to just do it where we inspect `Istio` and `IstioRevision` ?

Reply:
I forgot the reason why it's here. Didn't we discuss the option that IstioCNI will no longer be cluster-scoped resource in the future?

Reply:
No, IstioCNI and all others will always be cluster-scoped. But we did talk about maybe being able to integrate IstioCNI into the Istio resource when CNI starts supporting revisions (but this is currently not planned).

Reply:
Thanks, I will move it suggested location.